apps/aecore/src/aec_chain_state.erl

```diff
@@ -175,6 +176,17 @@ get_block_by_height(Height, ?assert_state() = State) when is_integer(Height),
             end
     end.
 
+-spec hash_is_connected_to_genesis(binary(), state()) -> boolean().
+hash_is_connected_to_genesis(Hash, ?assert_state() = State) when is_binary(Hash) ->
+    case blocks_db_find(Hash, State) of
+        error -> false;
+        {ok, Node} ->
+            try determine_chain_relation(Node, State) of
+                off_chain -> false;
+                _         -> true
```

Comment:
I see the alternatives are `in_chain | {fork, ForkNode} | new_top`. Open ended case may be error prone but it is ok.